### PR TITLE
Implementation of RStudio Electron command-line switches

### PR DESCRIPTION
**Pull request by matheustavaresdev**
### Intent

Improved RStudio Electron CLI support

### Approach

Created a file (`args-manager.ts`) that will manage every command that RStudio Electron can receive. Yargs was added for improved support of `--help` and `--version`

### Automated Tests

Changes on a single automated test were made on [application.test.ts](https://github.com/rstudio/rstudio/blob/2df9c4fb24a169b32d90648de45632d17e0db7e6/src/node/desktop/test/unit/main/application.test.ts#L42-L50) to comply with new code.

### QA Notes

Instructions on how to run were made on the [args-manager.ts](https://github.com/rstudio/rstudio/blob/908699036710d33847cf63dd6b57cbf310025923/src/node/desktop/src/main/args-manager.ts#L11-L16) file for when running with `npm` or with the `Mac binary`

### Checklist

- [X] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [X] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [X] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [X] This PR passes all local unit tests

Addresses #10448

### CODE REVIEW

**Comment by matheustavaresdev** [comment]
Here are some [examples](https://github.com/rstudio/rstudio/blob/fabfe8a8e605e4d5897e541954d7732fb30f8402/src/node/desktop/src/main/args-manager.ts#L11-L17) on how to run with `npm` and the Mac Binary. Not sure on how to run on windows and linux binaries though... If any of you can add that or comment so I can add on the file would be great! cc @gtritchie @timtmok @kevinushey

**Comment by gtritchie** [comment]
> Not sure on how to run on windows and linux binaries though... If any of you can add that or comment

@matheustavaresdev

On Windows, RStudio Electron is normally installed as `C:\Program Files\RStudio\rstudio.exe`.

On Linux, installed as `/usr/lib/rstudio/rstudio`, but we also create a link from `/usr/bin/rstudio` to that. So it's on the path and you can just run `rstudio` from terminal on Linux.

File: src/node/desktop/src/main/args-manager.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      name: kVersionJson,
+      describe: 'Report an extended version info',
+    },
+    {
+      name: kRunDiagnosticsOption,
+      describe: 'Run diagnostics and save in a file',
+    },
+    {
+      name: kLogLevel,
+      describe: 'Sets the verbosity of the logging',
```
Comment (gtritchie) [inline src/node/desktop/src/main/args-manager.ts:44]: Let's show the valid options: `--log-level=ERR|WARN|INFO|DEBUG`.

File: src/node/desktop/src/main/args-manager.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+// export const kSessionServerOption = '--session-server';
+// export const kSessionServerUrlOption = '--session-url';
+// export const kTempCookiesOption = '--use-temp-cookies';
+
+// !IMPORTANT: If some args should early exit the application, add them to `webpack.plugins.js`
+export class ArgsManager {
+  argsList = [
+    {
+      name: kVersionJson,
+      describe: 'Report an extended version info',
```
Comment (gtritchie) [inline src/node/desktop/src/main/args-manager.ts:36]: `Display versions of major components in JSON format`

**Review by gtritchie — CHANGES_REQUESTED** [review]

File: src/node/desktop/src/main/args-manager.ts (hunk 1 above)
Comment (matheustavaresdev) [reply src/node/desktop/src/main/args-manager.ts]: Fixed!

File: src/node/desktop/src/main/args-manager.ts (hunk 2 above)
Comment (matheustavaresdev) [reply src/node/desktop/src/main/args-manager.ts]: Fixed!

**Review by gtritchie — APPROVED** [review]
LGTM!

**Pull request merged**